src/rustfmt/main.rs

```diff
@@ -413,6 +418,10 @@ fn format_string(input: String, opt: Opt) -> Result<i32> {
         verbosity: Verbosity::Quiet,
     };
     let report = rustfmt_nightly::format(Input::Text(input), &config, setting)?;
+    if report.has_invalid_rustfmt_attribute_errors() && !opt.format_with_invalid_attributes {
+        eprintln!("Input contained invalid rustfmt attributes");
+        return Ok(1);
+    }
```

Comment:
I spent some debating with myself what the behavior should be on stdin input. I've opted for handling both file and stdin inputs the same (failing/not formatting in the presence of invalid rustfmt attributes unless the user explicitly opts in), but that also warranted the message on line 422 to avoid silently failing (since we don't emit the error details on stdin).

The other alternative I considered was still emitting the format report, but incorporating the attribute errors & flag for the determination of the exit code.